app/views/layouts/application.html.erb

```diff
@@ -16,7 +16,7 @@
 
       <div class="pull-right user-info">
         <% if current_user %>
-          <%= image_tag current_user.gravatar_url %> <%= current_user.name %> - <%= link_to "Sign Out", session_path(current_user), method: :delete %>
+          <%= image_tag avatar_url(current_user) %> <%= current_user.name %> - <%= link_to "Sign Out", session_path(current_user), method: :delete %>
```

Comment:
`avatar_url` was a helper method that you pass a user into, not an attribute on user